src/inject/dynamic-theme/stylesheet-modifier.ts

```diff
@@ -73,16 +81,18 @@ export function createStyleSheetModifier() {
 
             // Put CSS text with inserted CSS variables into separate <style> element
             // to properly handle composite properties (e.g. background -> background-color)
-            let vars: {sheet: CSSStyleSheet; remove: () => void};
+            let vars: CSSStyleSheet;
             let varsRule: CSSStyleRule = null;
+            let varIndex: number = null;
             if (variables.size > 0 || cssText.includes('var(')) {
                 const cssTextWithVariables = replaceCSSVariables(cssText, variables);
                 if (rulesTextCache.get(cssText) !== cssTextWithVariables) {
                     rulesTextCache.set(cssText, cssTextWithVariables);
                     textDiffersFromPrev = true;
                     vars = getTempCSSStyleSheet();
-                    vars.sheet.insertRule(cssTextWithVariables);
-                    varsRule = vars.sheet.cssRules[0] as CSSStyleRule;
+                    vars.insertRule(cssTextWithVariables);
+                    varIndex = vars.cssRules.length - 1;
```

Comment:
Not sure if we need dynamically `varIndex`. As this code should runs synchronous and can't be overridden mid-way by some MutationObserver, right?

Reply:
I think yes, there can be only a single rule.

Reply:
Great, removed it alongside re-using the CSSStyleSheet as well.